hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/mapreduce/HadoopPipelineMapper.java

```diff
@@ -0,0 +1,97 @@
+package org.molgenis.hadoop.pipeline.application.mapreduce;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.log4j.Logger;
+import org.molgenis.hadoop.pipeline.application.HadoopPipelineApplication;
+import org.molgenis.hadoop.pipeline.application.processes.BwaAligner;
+
+/**
+ * Hadoop MapReduce Job mapper.
+ */
+public class HadoopPipelineMapper extends Mapper<NullWritable, BytesWritable, NullWritable, Text> // SAMRecordWritable
+{
+	/**
+	 * Logger to write information to.
+	 */
+	private static final Logger logger = Logger.getLogger(HadoopPipelineMapper.class);
+
+	/**
+	 * Job context.
+	 */
+	Context context;
+
+	/**
+	 * BwaTool executable location.
+	 */
+	String bwaTool;
+	/**
+	 * Alignment reference fasta file location (with index file having the same prefix and being in the same directory.
+	 */
+	String alignmentReferenceFastaFile;
+
+	/**
+	 * BwaAligner instance to execute BWA alignment with.
+	 */
+	BwaAligner bwaAligner;
+
```

Comment:
Already implemented for these variables in the follow-up push, though it hasn't been tested as of yet (it contains multiple improvements to this and other classes so therefore hasn't been pushed yet).